Felienne/hedy
# [CLEAN-UP] Removes front-end dependency on javascript input validation

**Description**
In this PR we perform a major clean-up on how we approach and handle form data. In the current situation the form data is extracted with TypeScript by a function call `onsubmit`. This data should always conform the fixed `User` interface defined by TypeScript. This data is than sent to the back-end where the actual handling and verification is performed.

However, this can be simplified. The interface layer on TypeScript can be removed as the handling and checking of the data is already performed on the back-end. Therefore we replace this structure by an `onsubmit` trigger on the forms like this:

```
$('form#login').submit(function(e) {
 // Perform all stuff
});
```

Within this function we convert all form inputs found within the triggered submit form into a JSON object and sent this to the back-end. This process change has the advantage that we don't have to adjust an additional layer when changing the input fields. Currently we have to change the front-end input, the TypeScript interface and the back-end. In this PR we reduce this to changing the template and back-end.

Lastly we also clean-up the `auth.ts` overall. By removing redundant front-end session values that we no longer use and simplifying the redirecting process on several actions.

**Fixes**
This PR fixes #3063.

**How to test**
As this is a clean-up task everything should still work. Do **not** rely on the tests as they don't test the actual forms and templates. Therefore verify that all forms still work as expected with corresponding expected error messages. These are login, signup, profile update, public profile, logout, destroy, change password.

**Note**
We should test the reset and recover of the password both on Alpha. Due to the mail dependency we can't test these locally.
